raft/src/main/java/org/apache/kafka/raft/RaftClient.java

```diff
@@ -43,9 +45,10 @@
      * This method must be thread-safe.
      *
      * @param records The records to append to the log
+     * @param recordType control record type if any
      * @return A future containing the base offset and epoch of the appended records (if successful)
      */
-    CompletableFuture<OffsetAndEpoch> append(Records records);
+    CompletableFuture<OffsetAndEpoch> append(Records records, Optional<ControlRecordType> recordType);
```

Comment:
Related to previous comments, but I don't think we want to expose this in the public API. I think the raft server should own the control records. It shouldn't be possible for the state machine to send its own control records. At least until we come up with a good reason why they should.